### PR TITLE
Add use_named_constants to analysis options and fix errors

### PR DESCRIPTION
**Pull request by tarrinneal**
Added the use_named_constants line to analysis options and fixed all new violation errors.

*List which issues are fixed by this PR. You must list at least one issue.*
Some fixes for #96229

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter_android/CHANGELOG.md
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_android/CHANGELOG.md:1]: Only a couple of packages should have had a version change (not including this one). It looks like maybe your base-sha was determined incorrectly; it's supposed to Just Work, but some people have been having issues.

If you run
```
git co 00f464f4e1673698de939762a8a88cd138130ee3 -- packages/*/*/CHANGELOG.md packages/*/*/pubspec.yaml
git commit -m "revert metada changes"
dart script/tool/bin/flutter_plugin_tools.dart update-release-info --version=minimal --base-sha=00f464f4e1673698de939762a8a88cd138130ee3 --changelog='Fixes violations of new analysis option use_named_constants.'
```

it should come out right. I'll update the wiki to discuss `base-sha` since this has come up a few times now.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/google_maps_flutter/google_maps_flutter_android/CHANGELOG.md
Comment (tarrinneal) [reply packages/google_maps_flutter/google_maps_flutter_android/CHANGELOG.md:1]: I thought this seemed wrong, I'll get it fixed

**Comment by stuartmorgan** [comment]
Overriding the version change flag for `video_player`; the change is only to example/lib/main.dart and is so trivial it's not important for it to be updated on pub.dev.

File: packages/google_maps_flutter/google_maps_flutter/pubspec.yaml
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter/pubspec.yaml:7]: This is gone on main now, so you just need to merge in the latest to fix this. Sorry about that!

**Review by stuartmorgan — APPROVED** [review]
LGTM once it's merged to fix analyze

File: packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md:4]: This got merged incorrectly; you added this to the release notes of an already-released version, and didn't add a new version update to release the changed production code. (Normally CI would have caught this, but the override we added for video_player masked it).

Could you do a follow-up to move this line to a new 2.1.12?

Comment (tarrinneal) [reply packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md:4]: I'll send that up shortly

Comment (tarrinneal) [reply packages/google_maps_flutter/google_maps_flutter/CHANGELOG.md:4]: new PR here https://github.com/flutter/plugins/pull/6289

**Pull request merged**
